be/src/storage/lake/spill_mem_table_sink.cpp

```diff
@@ -37,7 +37,15 @@ Status LoadSpillOutputDataStream::append(RuntimeState* state, const std::vector<
     RETURN_IF_ERROR(_preallocate(total_size));
     // append data
     _append_bytes += total_size;
-    return _block->append(data);
+    auto st = _block->append(data);
+    if (st.is_capacity_limit_exceeded()) {
+        // No space left on device
+        // Try to acquire a new block from remote storage.
+        RETURN_IF_ERROR(_switch_to_remote_block(total_size));
+        st = _block->append(data);
+    }
+    _append_bytes += total_size;
```

Comment:
why do it again as line 39? should the `_append_bytes` be only updated when the st is ok?

Reply:
You are right, fixed.